### PR TITLE
RS-180: Web console shows license information

### PR DESCRIPTION
**Pull request by AnthonyCvn**
Closes #

### Please check if the PR fulfills these requirements

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] CHANGELOG.md has been updated (for bug fixes / features / docs)

### What kind of change does this PR introduce?

Feature

### What was changed?

A tab to show the license information.

Sample license:
license-info

No license:
no-license

Usage tab:
server-usage

### Related issues

No

### Does this PR introduce a breaking change?

Yes, it works only with reduct-js > 1.9.1

### Other information:

A bug fix was needed on reduct-js to use it on the browser. Please update to reduct-js > 1.9.1 to test the web console.

### CODE REVIEW

**Comment by AnthonyCvn** [comment]
@atimin I need a first review from you to check if the layout is good. I will write some tests once validated.

**Review by atimin — CHANGES_REQUESTED** [review]
I love the idea to use a tab for the license information. Looks great for me.

Few minor remarks:

1. Expire data looks strange on my side:
image

2. Maybe we could use bold font for names of field:

**Plan**: UNLIMITED

3. Don't forget that we need validation as well. See here the example: 

https://github.com/reductstore/reductstore/blob/main/reduct_cli/src/io/reduct.rs#L38

**Comment by AnthonyCvn** [comment]
@atimin everything should be good and tested :) Please have a look. Just a small point, I assumed the expiration date from the license to return the number of milliseconds since midnight, January 1, 1970 (UTC), I think it's the case, but just to double check.

**Comment by AnthonyCvn** [comment]
Screenshot 2024-03-22 at 16 32 05

File: src/Components/LicenseDetails/LicenseDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const expiryDate = new Date(license.expiryDate).toLocaleDateString(undefined,
+    {
+      year: "numeric",
+      month: "long",
+      day: "numeric",
+    }
+  );
+
+  const hasExpired = Date.now() > license.expiryDate;
+  const isOverQuota = usage > BigInt(license.diskQuota);
```
Comment (atimin) [inline src/Components/LicenseDetails/LicenseDetails.tsx:24]: 0 means unlimited.

Comment (atimin) [reply src/Components/LicenseDetails/LicenseDetails.tsx:24]: I think we could even show it in Disk Quota field.

Comment (atimin) [reply src/Components/LicenseDetails/LicenseDetails.tsx:24]: And it is in terabytes

File: src/Components/LicenseDetails/LicenseDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const alertExplanation = hasExpired ? "Your license has expired. " : isOverQuota ? "You have exceeded your disk quota. " : "";
+
+  return (
+    <>
+      {alertExplanation && <div className="licenseAlert"><LicenseAlert alertExplanation={alertExplanation} /></div>}
```
Comment (atimin) [inline src/Components/LicenseDetails/LicenseDetails.tsx:29]: image

I'm not sure if the mentioning of BUSL is relevant here. Maybe we could say that they must contact us to extend it.

**Review by atimin — CHANGES_REQUESTED** [review]
Looks great but there are mistakes. See my comments. 

Is it possible to add an icon the License tab header  when it is BUSL or expired. So the users can see that there is a problem without clicking on the tab.

**Review by atimin — APPROVED** [review]
Perfect!

**Pull request merged**